workshop_sections/word2vec/README.md

```diff
@@ -1,120 +1,178 @@
+This will run your model in 5 processes (2 parameter servers and 2 workers). You will see output from all 5 in your terminal.
 
-### From TFRecords
+### Training in the Cloud with the Google Cloud Machine Learning API
 
-If you don't want to do preprocessing in memory, or you want to preprocess via another distributed graph computation framework, Tensorflow supports reading from a binary (protobuf) format called TFRecords. We have provided this file publicly available online. See below for how to train with this data.
+To train in the Cloud, we use `gcloud` to submit a training job. It's important to note that here we must use a Google Cloud Storage output directory. Make sure you have set up a Google Cloud Storage bucket and authorized the Google Cloud ML API to access it. See instructions [here](../../INSTALL.md#project-and-cloud-ml-setup).
 
-## Training the model
+After you have completed this, set `MY_BUCKET=gs://my-bucket` with your name, for the commands below.
 
-To define your model, run:
+Then submit a training job with:
 
 ```
-import tensorflow as tf
-import os
-import time
-workshop_root = os.path.dirname(os.path.dirname(os.getcwd()))
-output_dir = os.path.join(workshop_root, 'workshop-data', 'output', str(int(time.time())))
-tf.logging.set_verbosity(tf.logging.INFO)
-word2vec_model = tf.contrib.learn.Estimator(model_fn=word2vec.make_model_fn(index, counts.tolist()), model_dir=output_dir)
+gcloud beta ml jobs submit training myword2vectraining \
+  --staging-bucket $MY_BUCKET \
+  --module-name word2vec.task \
+  --package-path word2vec \
+  --config config.yaml \
+  -- \
+  --output-path $MY_BUCKET/word2vec/output/run1 \
+  --vocab-file gs://tf-ml-workshop/word2vec/data/text8-vocab.tsv \
+  --train-data-file gs://tf-ml-workshop/word2vec/data/text8-train.pb2 \
+  --eval-data-file gs://tf-ml-workshop/word2vec/data/text8-eval.pb2
 ```
 
-To train with the in memory numpy arrays, run:
+This will output TensorBoard summaries, and model checkpoints to `$MY_BUCKET/word2vec/output/run1`
 
-```
-word2vec_model.fit(x=targets, y=contexts, batch_size=256, max_steps=int(len(targets)/256))
-```
+Additional configuration can be added in the `config.yaml` file. For the schema of this file check out the [Google Cloud ML Docs](https://cloud.google.com/ml/reference/configuration-data-structures#yaml_configuration_file)
+
+## Viewing Embeddings With TensorBoard
+
+To view embeddings (and other summaries) with TensorBoard, simply start TensorBoard pointing it at your output directory (in the Cloud or locally). For example:
 
-To train with the pre-preprocessed TFRecords, run:
 ```
-word2vec_model.fit(input_fn=word2vec.input_from_files(word2vec.GCS_SKIPGRAMS, 256))
+tensorboard --logdir $MY_BUCKET/word2vec/output/run1
```

Comment:
If you should change how the output dir is generated (as per my comment above), you'll want to edit this too.